contracts/Proxy/TimeOwnedUpgradeabilityProxy.sol

```diff
@@ -0,0 +1,45 @@
+pragma solidity ^0.5.13;
+
+import { OwnedUpgradeabilityProxy } from "./OwnedUpgradeabilityProxy.sol";
+
+/**
+ * @title TimeOwnedUpgradeabilityProxy
+ * @dev This contract combines an upgradeability proxy with 
+ * basic authorization control functionalities
+ *
+ * This contract allows us to specify a time at which the proxy can no longer
+ * be upgraded
+ */
+contract TimeOwnedUpgradeabilityProxy is OwnedUpgradeabilityProxy {
+
+    uint256 expiration;
+    uint256 constant MAX_UINT = 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff;
+
+    /**
+    * @dev the constructor sets the original owner of the contract to the sender account.
+    */
+    constructor() public {
+        _setUpgradeabilityOwner(msg.sender);
+        expiration = MAX_UINT;
+    }
+
+    /**
+     *
+    */
+    function setExpiration(uint256 _newTimestamp) external onlyProxyOwner {
```

Comment:
Idea: perhaps from compliance/legal perspective it would be better to initialise expiration to in constructor (eg. +3months) and instead of `setExpiration` have `prolongExpiration`, which would change the `expiration` by a fixed value (eg. 1 month). The drawback of this solution is that we cannot "turn off" upgradability, which we can currently by setting `expiration` to a timestamp in the past. Anyway, it's your call @RafaelCosman to leave it as it is now or change it.